lib/services/presets_test.go

```diff
@@ -782,3 +785,49 @@ func TestAddRoleDefaults(t *testing.T) {
 		})
 	}
 }
+
+func TestPresetRolesDumped(t *testing.T) {
+	// This test ensures that the most recent version of selected preset roles
+	// have been correctly dumped to a generated JSON file. We use a generated
+	// file, because it's simpler to load it from a TypeScript test this way,
+	// rather than calling a Go binary.
+
+	// First, get the required roles, as defined in our codebase, and set their
+	// defaults.
+	access := NewPresetAccessRole()
+	editor := NewPresetEditorRole()
+	auditor := NewPresetAuditorRole()
+	rolesByName := map[string]types.Role{
+		access.GetName():  access,
+		editor.GetName():  editor,
+		auditor.GetName(): auditor,
+	}
+	for _, r := range rolesByName {
+		err := CheckAndSetDefaults(r)
+		require.NoError(t, err)
+	}
+
+	// Next, dump them all to JSON and parse them back again. This step is
+	// necessary, because unmarshaling isn't precisely the opposite of
+	// marshaling, and comparing raw roles to their unmarshaled counterparts
+	// still lead to some discrepancies. We can't also directly compare JSON
+	// blobs, as it's hard to reason whether this process is entirely
+	// deterministic.
+	bytes, err := json.Marshal(rolesByName)
+	require.NoError(t, err)
+	var recreatedRolesByName map[string]types.RoleV6
+	err = json.Unmarshal(bytes, &recreatedRolesByName)
+	require.NoError(t, err)
+
+	// Read the roles defined in the generated file.
+	bytes, err = os.ReadFile("../../gen/preset-roles.json")
+	require.NoError(t, err)
+	var rolesFromFile map[string]types.RoleV6
+	err = json.Unmarshal(bytes, &rolesFromFile)
+	require.NoError(t, err)
+
+	// Finally, compare the roles.
+	assert.Equal(t, rolesFromFile, recreatedRolesByName,
```

Comment:
I think assert is not necessarily needed here, right?

```suggestion
	require.Equal(t, rolesFromFile, recreatedRolesByName,
```

The assert import has to be dropped after this change.